### PR TITLE
feat(clerk-expo): Support expo passkeys

### PR DESCRIPTION
**Pull request by AlexNti**
## Description

 🔐  This PR introduces passkey support for Expo iOS, Android, and Web.  🚀

##  🛠 Usage

To enable passkeys in your Expo project, update your `ClerkProvider` as follows:

```tsx
import { ClerkProvider } from '@clerk/clerk-expo';
import { passkeys } from '@clerk/clerk-expo/passkeys';

<ClerkProvider __experimental__passkeys={passkeys}>
  {/* Your app here */}
</ClerkProvider>
```
### 🔑 Creating a Passkey

```tsx
const { user } = useUser();

const handleCreatePasskey = async () => {
  if (!user) return;
  try {
    return await user.createPasskey();
  } catch (e: any) {
    // handle error
  }
};
```

### 🔓 Authenticating with a Passkey

```tsx
const { signIn, setActive } = useSignIn();

const handlePasskeySignIn = async () => {
  try {
    const signInResponse = await signIn.authenticateWithPasskey();
    await setActive({ session: signInResponse.createdSessionId });
  } catch (err: any) {
    // handle error
  }
};
```

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

**Comment by AlexNti** [comment]
!snapshot

**Comment by AlexNti** [comment]
!snapshot

**Comment by AlexNti** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @AlexNti - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/astro | 1.3.16-snapshot.v84a6da3 |
| @clerk/backend | 1.14.1-snapshot.v84a6da3 |
| @clerk/express | 1.3.1-snapshot.v84a6da3 |
| @clerk/fastify | 2.0.3-snapshot.v84a6da3 |
| @clerk/nextjs | 5.7.5-snapshot.v84a6da3 |
| @clerk/remix | 4.2.39-snapshot.v84a6da3 |
| @clerk/clerk-sdk-node | 5.0.52-snapshot.v84a6da3 |
| @clerk/tanstack-start | 0.4.15-snapshot.v84a6da3 |
| @clerk/testing | 1.3.13-snapshot.v84a6da3 |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/astro`
```sh
npm i @clerk/astro@1.3.16-snapshot.v84a6da3 --save-exact
```
`@clerk/backend`
```sh
npm i @clerk/backend@1.14.1-snapshot.v84a6da3 --save-exact
```
`@clerk/express`
```sh
npm i @clerk/express@1.3.1-snapshot.v84a6da3 --save-exact
```
`@clerk/fastify`
```sh
npm i @clerk/fastify@2.0.3-snapshot.v84a6da3 --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@5.7.5-snapshot.v84a6da3 --save-exact
```
`@clerk/remix`
```sh
npm i @clerk/remix@4.2.39-snapshot.v84a6da3 --save-exact
```
`@clerk/clerk-sdk-node`
```sh
npm i @clerk/clerk-sdk-node@5.0.52-snapshot.v84a6da3 --save-exact
```
`@clerk/tanstack-start`
```sh
npm i @clerk/tanstack-start@0.4.15-snapshot.v84a6da3 --save-exact
```
`@clerk/testing`
```sh
npm i @clerk/testing@1.3.13-snapshot.v84a6da3 --save-exact
```

**Comment by anagstef** [comment]
!snapshot

**Comment by AlexNti** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @AlexNti - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/astro | 1.4.1-snapshot.v1f5c4b4 |
| @clerk/backend | 1.15.1-snapshot.v1f5c4b4 |
| @clerk/chrome-extension | 1.3.22-snapshot.v1f5c4b4 |
| @clerk/clerk-js | 5.29.0-snapshot.v1f5c4b4 |
| @clerk/elements | 0.17.1-snapshot.v1f5c4b4 |
| @clerk/clerk-expo | 2.3.0-snapshot.v1f5c4b4 |
| @clerk/express | 1.3.3-snapshot.v1f5c4b4 |
| @clerk/fastify | 2.0.5-snapshot.v1f5c4b4 |
| @clerk/localizations | 3.3.2-snapshot.v1f5c4b4 |
| @clerk/nextjs | 6.0.1-snapshot.v1f5c4b4 |
| @clerk/clerk-react | 5.13.1-snapshot.v1f5c4b4 |
| @clerk/remix | 4.2.41-snapshot.v1f5c4b4 |
| @clerk/clerk-sdk-node | 5.0.54-snapshot.v1f5c4b4 |
| @clerk/shared | 2.10.1-snapshot.v1f5c4b4 |
| @clerk/tanstack-start | 0.4.17-snapshot.v1f5c4b4 |
| @clerk/testing | 1.3.15-snapshot.v1f5c4b4 |
| @clerk/themes | 2.1.39-snapshot.v1f5c4b4 |
| @clerk/types | 4.28.0-snapshot.v1f5c4b4 |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/astro`
```sh
npm i @clerk/astro@1.4.1-snapshot.v1f5c4b4 --save-exact
```
`@clerk/backend`
```sh
npm i @clerk/backend@1.15.1-snapshot.v1f5c4b4 --save-exact
```
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.3.22-snapshot.v1f5c4b4 --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.29.0-snapshot.v1f5c4b4 --save-exact
```
`@clerk/elements`
```sh
npm i @clerk/elements@0.17.1-snapshot.v1f5c4b4 --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@2.3.0-snapshot.v1f5c4b4 --save-exact
```
`@clerk/express`
```sh
npm i @clerk/express@1.3.3-snapshot.v1f5c4b4 --save-exact
```
`@clerk/fastify`
```sh
npm i @clerk/fastify@2.0.5-snapshot.v1f5c4b4 --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@3.3.2-snapshot.v1f5c4b4 --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@6.0.1-snapshot.v1f5c4b4 --save-exact
```
`@clerk/clerk-react`
```sh
npm i @clerk/clerk-react@5.13.1-snapshot.v1f5c4b4 --save-exact
```
`@clerk/remix`
```sh
npm i @clerk/remix@4.2.41-snapshot.v1f5c4b4 --save-exact
```
`@clerk/clerk-sdk-node`
```sh
npm i @clerk/clerk-sdk-node@5.0.54-snapshot.v1f5c4b4 --save-exact
```
`@clerk/shared`
```sh
npm i @clerk/shared@2.10.1-snapshot.v1f5c4b4 --save-exact
```
`@clerk/tanstack-start`
```sh
npm i @clerk/tanstack-start@0.4.17-snapshot.v1f5c4b4 --save-exact
```
`@clerk/testing`
```sh
npm i @clerk/testing@1.3.15-snapshot.v1f5c4b4 --save-exact
```
`@clerk/themes`
```sh
npm i @clerk/themes@2.1.39-snapshot.v1f5c4b4 --save-exact
```
`@clerk/types`
```sh
npm i @clerk/types@4.28.0-snapshot.v1f5c4b4 --save-exact
```

**Comment by AlexNti** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @AlexNti - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/astro | 1.4.1-snapshot.va96a299 |
| @clerk/backend | 1.15.1-snapshot.va96a299 |
| @clerk/chrome-extension | 1.3.22-snapshot.va96a299 |
| @clerk/clerk-js | 5.29.0-snapshot.va96a299 |
| @clerk/elements | 0.17.1-snapshot.va96a299 |
| @clerk/clerk-expo | 2.3.0-snapshot.va96a299 |
| @clerk/express | 1.3.3-snapshot.va96a299 |
| @clerk/fastify | 2.0.5-snapshot.va96a299 |
| @clerk/localizations | 3.3.2-snapshot.va96a299 |
| @clerk/nextjs | 6.0.1-snapshot.va96a299 |
| @clerk/clerk-react | 5.13.1-snapshot.va96a299 |
| @clerk/remix | 4.2.41-snapshot.va96a299 |
| @clerk/clerk-sdk-node | 5.0.54-snapshot.va96a299 |
| @clerk/shared | 2.10.1-snapshot.va96a299 |
| @clerk/tanstack-start | 0.4.17-snapshot.va96a299 |
| @clerk/testing | 1.3.15-snapshot.va96a299 |
| @clerk/themes | 2.1.39-snapshot.va96a299 |
| @clerk/types | 4.28.0-snapshot.va96a299 |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/astro`
```sh
npm i @clerk/astro@1.4.1-snapshot.va96a299 --save-exact
```
`@clerk/backend`
```sh
npm i @clerk/backend@1.15.1-snapshot.va96a299 --save-exact
```
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.3.22-snapshot.va96a299 --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.29.0-snapshot.va96a299 --save-exact
```
`@clerk/elements`
```sh
npm i @clerk/elements@0.17.1-snapshot.va96a299 --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@2.3.0-snapshot.va96a299 --save-exact
```
`@clerk/express`
```sh
npm i @clerk/express@1.3.3-snapshot.va96a299 --save-exact
```
`@clerk/fastify`
```sh
npm i @clerk/fastify@2.0.5-snapshot.va96a299 --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@3.3.2-snapshot.va96a299 --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@6.0.1-snapshot.va96a299 --save-exact
```
`@clerk/clerk-react`
```sh
npm i @clerk/clerk-react@5.13.1-snapshot.va96a299 --save-exact
```
`@clerk/remix`
```sh
npm i @clerk/remix@4.2.41-snapshot.va96a299 --save-exact
```
`@clerk/clerk-sdk-node`
```sh
npm i @clerk/clerk-sdk-node@5.0.54-snapshot.va96a299 --save-exact
```
`@clerk/shared`
```sh
npm i @clerk/shared@2.10.1-snapshot.va96a299 --save-exact
```
`@clerk/tanstack-start`
```sh
npm i @clerk/tanstack-start@0.4.17-snapshot.va96a299 --save-exact
```
`@clerk/testing`
```sh
npm i @clerk/testing@1.3.15-snapshot.va96a299 --save-exact
```
`@clerk/themes`
```sh
npm i @clerk/themes@2.1.39-snapshot.va96a299 --save-exact
```
`@clerk/types`
```sh
npm i @clerk/types@4.28.0-snapshot.va96a299 --save-exact
```

**Comment by AlexNti** [comment]
!snapshot

**Comment by AlexNti** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @AlexNti - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/astro | 1.4.1-snapshot.vb92f3ea |
| @clerk/backend | 1.15.1-snapshot.vb92f3ea |
| @clerk/chrome-extension | 1.3.22-snapshot.vb92f3ea |
| @clerk/clerk-js | 5.29.0-snapshot.vb92f3ea |
| @clerk/elements | 0.17.1-snapshot.vb92f3ea |
| @clerk/clerk-expo | 2.3.0-snapshot.vb92f3ea |
| @clerk/express | 1.3.3-snapshot.vb92f3ea |
| @clerk/fastify | 2.0.5-snapshot.vb92f3ea |
| @clerk/localizations | 3.3.2-snapshot.vb92f3ea |
| @clerk/nextjs | 6.0.1-snapshot.vb92f3ea |
| @clerk/clerk-react | 5.13.1-snapshot.vb92f3ea |
| @clerk/remix | 4.2.41-snapshot.vb92f3ea |
| @clerk/clerk-sdk-node | 5.0.54-snapshot.vb92f3ea |
| @clerk/shared | 2.10.1-snapshot.vb92f3ea |
| @clerk/tanstack-start | 0.4.17-snapshot.vb92f3ea |
| @clerk/testing | 1.3.15-snapshot.vb92f3ea |
| @clerk/themes | 2.1.39-snapshot.vb92f3ea |
| @clerk/types | 4.28.0-snapshot.vb92f3ea |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/astro`
```sh
npm i @clerk/astro@1.4.1-snapshot.vb92f3ea --save-exact
```
`@clerk/backend`
```sh
npm i @clerk/backend@1.15.1-snapshot.vb92f3ea --save-exact
```
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.3.22-snapshot.vb92f3ea --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.29.0-snapshot.vb92f3ea --save-exact
```
`@clerk/elements`
```sh
npm i @clerk/elements@0.17.1-snapshot.vb92f3ea --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@2.3.0-snapshot.vb92f3ea --save-exact
```
`@clerk/express`
```sh
npm i @clerk/express@1.3.3-snapshot.vb92f3ea --save-exact
```
`@clerk/fastify`
```sh
npm i @clerk/fastify@2.0.5-snapshot.vb92f3ea --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@3.3.2-snapshot.vb92f3ea --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@6.0.1-snapshot.vb92f3ea --save-exact
```
`@clerk/clerk-react`
```sh
npm i @clerk/clerk-react@5.13.1-snapshot.vb92f3ea --save-exact
```
`@clerk/remix`
```sh
npm i @clerk/remix@4.2.41-snapshot.vb92f3ea --save-exact
```
`@clerk/clerk-sdk-node`
```sh
npm i @clerk/clerk-sdk-node@5.0.54-snapshot.vb92f3ea --save-exact
```
`@clerk/shared`
```sh
npm i @clerk/shared@2.10.1-snapshot.vb92f3ea --save-exact
```
`@clerk/tanstack-start`
```sh
npm i @clerk/tanstack-start@0.4.17-snapshot.vb92f3ea --save-exact
```
`@clerk/testing`
```sh
npm i @clerk/testing@1.3.15-snapshot.vb92f3ea --save-exact
```
`@clerk/themes`
```sh
npm i @clerk/themes@2.1.39-snapshot.vb92f3ea --save-exact
```
`@clerk/types`
```sh
npm i @clerk/types@4.28.0-snapshot.vb92f3ea --save-exact
```

**Comment by AlexNti** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @AlexNti - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/astro | 1.4.1-snapshot.vaa1630d |
| @clerk/backend | 1.15.1-snapshot.vaa1630d |
| @clerk/chrome-extension | 1.3.22-snapshot.vaa1630d |
| @clerk/clerk-js | 5.29.0-snapshot.vaa1630d |
| @clerk/elements | 0.17.1-snapshot.vaa1630d |
| @clerk/clerk-expo | 2.3.0-snapshot.vaa1630d |
| @clerk/express | 1.3.3-snapshot.vaa1630d |
| @clerk/fastify | 2.0.5-snapshot.vaa1630d |
| @clerk/localizations | 3.3.2-snapshot.vaa1630d |
| @clerk/nextjs | 6.0.1-snapshot.vaa1630d |
| @clerk/clerk-react | 5.13.1-snapshot.vaa1630d |
| @clerk/remix | 4.2.41-snapshot.vaa1630d |
| @clerk/clerk-sdk-node | 5.0.54-snapshot.vaa1630d |
| @clerk/shared | 2.10.1-snapshot.vaa1630d |
| @clerk/tanstack-start | 0.4.17-snapshot.vaa1630d |
| @clerk/testing | 1.3.15-snapshot.vaa1630d |
| @clerk/themes | 2.1.39-snapshot.vaa1630d |
| @clerk/types | 4.28.0-snapshot.vaa1630d |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/astro`
```sh
npm i @clerk/astro@1.4.1-snapshot.vaa1630d --save-exact
```
`@clerk/backend`
```sh
npm i @clerk/backend@1.15.1-snapshot.vaa1630d --save-exact
```
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.3.22-snapshot.vaa1630d --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.29.0-snapshot.vaa1630d --save-exact
```
`@clerk/elements`
```sh
npm i @clerk/elements@0.17.1-snapshot.vaa1630d --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@2.3.0-snapshot.vaa1630d --save-exact
```
`@clerk/express`
```sh
npm i @clerk/express@1.3.3-snapshot.vaa1630d --save-exact
```
`@clerk/fastify`
```sh
npm i @clerk/fastify@2.0.5-snapshot.vaa1630d --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@3.3.2-snapshot.vaa1630d --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@6.0.1-snapshot.vaa1630d --save-exact
```
`@clerk/clerk-react`
```sh
npm i @clerk/clerk-react@5.13.1-snapshot.vaa1630d --save-exact
```
`@clerk/remix`
```sh
npm i @clerk/remix@4.2.41-snapshot.vaa1630d --save-exact
```
`@clerk/clerk-sdk-node`
```sh
npm i @clerk/clerk-sdk-node@5.0.54-snapshot.vaa1630d --save-exact
```
`@clerk/shared`
```sh
npm i @clerk/shared@2.10.1-snapshot.vaa1630d --save-exact
```
`@clerk/tanstack-start`
```sh
npm i @clerk/tanstack-start@0.4.17-snapshot.vaa1630d --save-exact
```
`@clerk/testing`
```sh
npm i @clerk/testing@1.3.15-snapshot.vaa1630d --save-exact
```
`@clerk/themes`
```sh
npm i @clerk/themes@2.1.39-snapshot.vaa1630d --save-exact
```
`@clerk/types`
```sh
npm i @clerk/types@4.28.0-snapshot.vaa1630d --save-exact
```

File: packages/clerk-js/src/core/clerk.ts [hunk 1]
```diff
@@ -190,6 +195,24 @@ export class Clerk implements ClerkInterface {
...
+    | (({
+        publicKeyOptions,
+      }: {
+        publicKeyOptions: PublicKeyCredentialRequestOptionsWithoutExtensions;
+      }) => Promise<CredentialReturn<PublicKeyCredentialWithAuthenticatorAssertionResponse>>)
+    | undefined;
+
+  public __unstable__isWebAuthnSupported: (() => boolean) | undefined;
+  public __unstable__isWebAuthnAutofillSupported: (() => Promise<boolean>) | undefined;
+  public __unstable__isWebAuthnPlatformAuthenticatorSupported: (() => Promise<boolean>) | undefined;
```
Comment (anagstef) [inline packages/clerk-js/src/core/clerk.ts:214]: We should go for `__experimental_` instead of the `unstable` keyword!

Comment (anagstef) [inline packages/clerk-js/src/core/clerk.ts:214]: e.g.
```suggestion
  public __experimental_createPublicCredentials:
    | ((
        publicKey: PublicKeyCredentialCreationOptionsWithoutExtensions,
      ) => Promise<CredentialReturn<PublicKeyCredentialWithAuthenticatorAttestationResponse>>)
    | undefined;

  public __experimental_getPublicCredentials:
    | (({
        publicKeyOptions,
      }: {
        publicKeyOptions: PublicKeyCredentialRequestOptionsWithoutExtensions;
      }) => Promise<CredentialReturn<PublicKeyCredentialWithAuthenticatorAssertionResponse>>)
    | undefined;

  public __experimental_isWebAuthnSupported: (() => boolean) | undefined;
  public __experimental_isWebAuthnAutofillSupported: (() => Promise<boolean>) | undefined;
  public __experimental_isWebAuthnPlatformAuthenticatorSupported: (() => Promise<boolean>) | undefined;
```

File: packages/clerk-js/src/core/resources/SignIn.ts [hunk 2]
```diff
@@ -303,7 +303,13 @@ export class SignIn extends BaseResource implements SignInResource {
      * The UI should always prevent from this method being called if WebAuthn is not supported.
      * As a precaution we need to check if WebAuthn is supported.
      */
-    if (!isWebAuthnSupported()) {
+
+    const _isWebAuthnSupported = SignIn.clerk.__unstable__isWebAuthnSupported || isWebAuthnSupported;
```
Comment (anagstef) [inline packages/clerk-js/src/core/resources/SignIn.ts:307]: To make it more readable I would prefer that we rename on import and keep the initial value name without the underscore. 
Example:
```
import { isWebAuthnSupported as isWebAuthnSupportedOnWindow } from '@clerk/shared/webauthn';

// and then
const isWebAuthnSupported = SignIn.clerk.__unstable__isWebAuthnSupported || isWebAuthnSupportedOnWindow;
```

File: packages/clerk-js/src/core/resources/SignIn.ts [hunk 3]
```diff
@@ -303,7 +303,13 @@ export class SignIn extends BaseResource implements SignInResource {
...
+    const _webAuthnGetCredential = SignIn.clerk.__unstable__getPublicCredentials || webAuthnGetCredential;
+    const _isWebAuthnAutofillSupported =
+      SignIn.clerk.__unstable__isWebAuthnAutofillSupported || isWebAuthnAutofillSupported;
```
Comment (anagstef) [inline packages/clerk-js/src/core/resources/SignIn.ts:310]: Same as above!

File: packages/expo/src/provider/singleton/createClerkInstance.ts [hunk 4]
```diff
@@ -38,6 +43,46 @@ export function createClerkInstance(ClerkClass: typeof Clerk) {
       const saveToken = tokenCache.saveToken;
       __internal_clerk = clerk = new ClerkClass(publishableKey);
 
+      if (Platform.OS === 'ios' || Platform.OS === 'android') {
+        // @ts-expect-error - This is an internal API
+        __internal_clerk.__unstable__createPublicCredentials = (
```
Comment (anagstef) [inline packages/expo/src/provider/singleton/createClerkInstance.ts:48]: Also change these with the `__experimental_`

File: packages/expo/src/provider/singleton/createClerkInstance.ts [hunk 5]
```diff
@@ -38,6 +43,46 @@ export function createClerkInstance(ClerkClass: typeof Clerk) {
...
+        // @ts-expect-error - This is an internal API
+        __internal_clerk.__unstable__isWebAuthnAutofillSupported = () => {
+          return options?.passkeys?.isAutoFillSupported
+            ? options?.passkeys?.isAutoFillSupported()
+            : errorThrower.throw('isSupported() for passkeys is missing');
+        };
+
+        // @ts-expect-error - This is an internal API
+        __internal_clerk.__unstable__isWebAuthnPlatformAuthenticatorSupported = () => {
+          return Promise.resolve(true);
```
Comment (anagstef) [inline packages/expo/src/provider/singleton/createClerkInstance.ts:82]: Is this always true?

Comment (AlexNti) [reply packages/expo/src/provider/singleton/createClerkInstance.ts:82]: For devices will always be true.

File: packages/expo/src/provider/ClerkProvider.tsx [hunk 6]
```diff
@@ -6,13 +6,15 @@ import * as WebBrowser from 'expo-web-browser';
...
 import { getClerkInstance } from './singleton';
+import type { BuildClerkOptions } from './singleton/types';
 
 export type ClerkProviderProps = React.ComponentProps<typeof ClerkReactProvider> & {
   /**
    * The token cache is used to persist the active user's session token. Clerk stores this token in memory by default, however it is recommended to use a token cache for production applications.
    * @see https://clerk.com/docs/quickstarts/expo#configure-the-token-cache-with-expo
    */
   tokenCache?: TokenCache;
+  passkeys?: BuildClerkOptions['passkeys'];
```
Comment (anagstef) [inline packages/expo/src/provider/ClerkProvider.tsx:17]: Also let's add a comment here that mentions that this is experimental feature etc.

**Comment by AlexNti** [comment]
> We should use the `__experimental_` prefix because we don't use the `__unstable_` anymore!
> 
> Great job! 🚀

 Thank you for the review @anagstef , how about instead of renaming them from `unstable` to `internal`?

**Comment by anagstef** [comment]
> > We should use the `__experimental_` prefix because we don't use the `__unstable_` anymore!
> > Great job! 🚀
> 
> Thank you for the review @anagstef , how about instead of renaming them from `unstable` to `internal`?

Yes, that's even better! Let's go with `internal` instead of `unstable`, and use the `experimental` on the public-facing props!

File: .changeset/late-camels-talk.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
+---
+"@clerk/clerk-js": minor
+"@clerk/shared": minor
+"@clerk/types": minor
+"@clerk/clerk-expo": minor
+"@clerk/expo-passkeys": patch
+---
+
+Introduce support for passkeys in Expo (iOS, Android, and Web).
```
Comment (anagstef) [inline .changeset/late-camels-talk.md:9]:
```suggestion
Introduce expoerimental support for passkeys in Expo (iOS, Android, and Web).
```

File: .changeset/late-camels-talk.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+To use passkeys in Expo projects, pass the `__experimental__passkeys` object, which can be imported from `@clerk/clerk-expo/passkeys`, to the `ClerkProvider` component:
+
+```tsx
+
+import { ClerkProvider } from '@clerk/clerk-expo';
+import { passkeys } from '@clerk/clerk-expo/passkeys';
+
+<ClerkProvider __experimental__passkeys={passkeys}>
```
Comment (anagstef) [inline .changeset/late-camels-talk.md:18]: `__experimental_` needs a single underscore.
```suggestion
<ClerkProvider __experimental_passkeys={passkeys}>
```

File: packages/clerk-js/src/core/clerk.ts [hunk 9]
```diff
@@ -181,6 +186,24 @@ export class Clerk implements ClerkInterface {
   #pageLifecycle: ReturnType<typeof createPageLifecycle> | null = null;
   #touchThrottledUntil = 0;
 
+  public __internal__createPublicCredentials:
```
Comment (anagstef) [inline packages/clerk-js/src/core/clerk.ts:189]: Also, `__internal_XXX` needs a single underscore before the variable name.
```suggestion
  public __internal_createPublicCredentials:
```

**Review by anagstef — CHANGES_REQUESTED** [review]
We need to convert:
- `__experimental__XXX` to `__experimental_XXX`
- `__internal__XXX` to `__internal_XXX`

As our convention uses 1 underscore after the `experimental/internal` word.

File: packages/expo/src/provider/singleton/types.ts [hunk 10]
```diff
@@ -1,6 +1,30 @@
...
+
 import type { TokenCache } from '../../cache/types';
 
 export type BuildClerkOptions = {
   publishableKey?: string;
   tokenCache?: TokenCache;
+  /**
+   * This is an experimental prop that is being used from expo apps,
+   * this feature is only available for private beta users.
+   */
```
Comment (anagstef) [inline packages/expo/src/provider/singleton/types.ts:17]:
```suggestion
  /**
   * Note: Passkey support in Expo is currently in a limited rollout phase. 
   * If you're interested in using this feature, please contact us for early access or additional details.
   * 
   * @experimental This API is experimental and may change at any moment.
   */
```

File: packages/expo/src/provider/ClerkProvider.tsx [hunk 11]
```diff
@@ -6,13 +6,15 @@ import * as WebBrowser from 'expo-web-browser';
...
 import { getClerkInstance } from './singleton';
+import type { BuildClerkOptions } from './singleton/types';
 
 export type ClerkProviderProps = React.ComponentProps<typeof ClerkReactProvider> & {
   /**
    * The token cache is used to persist the active user's session token. Clerk stores this token in memory by default, however it is recommended to use a token cache for production applications.
    * @see https://clerk.com/docs/quickstarts/expo#configure-the-token-cache-with-expo
    */
   tokenCache?: TokenCache;
+  __experimental_passkeys?: BuildClerkOptions['__experimental_passkeys'];
```
Comment (anagstef) [inline packages/expo/src/provider/ClerkProvider.tsx:17]:
```suggestion
   * @see https://clerk.com/docs/quickstarts/expo#configure-the-token-cache-with-expo
   */
  tokenCache?: TokenCache;
   /**
   * Note: Passkey support in Expo is currently in a limited rollout phase. 
   * If you're interested in using this feature, please contact us for early access or additional details.
   * 
   * @experimental This API is experimental and may change at any moment.
   */
  __experimental_passkeys?: BuildClerkOptions['__experimental_passkeys'];
```

**Review by anagstef — APPROVED** [review]

**Pull request merged**